[Bransonlj] iP

Bransonlj opened the pull request:
# DUKE
Duke's functions:
- Add tasks
- See tasks
- Delete tasks

How to use:
1. Run Duke
2. ???
3. You have used Duke :tada:

```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Ui.class, args);
    }
}
```

Features:
- [x] Manage Tasks
- [ ] ChatGPDuke


Happy ~~Lunar~~ *Chinese* New Year `NTU.xd()`
[Free ang bao](https://www.google.com/search?q=ang+bao&rlz=1C1CHZN_enSG961SG961&sxsrf=AJOqlzXPJkDI5EqoRSSxJ8iKfcC0UV2Ezw:1675182922313&source=lnms&tbm=isch&sa=X&ved=2ahUKEwidupLYnvL8AhXN5XMBHVcfCVkQ_AUoAXoECAEQAw&biw=1835&bih=679&dpr=1.88)

> nian nian you yu :fish: -Confusius

Commit:
Add Gradle support

Bransonlj committed:
Level-1

Bransonlj committed:
Level-2: added list

Bransonlj committed:
Level-3 added Task class, allow marking and unmarking of tasks

Bransonlj committed:
Level-4 Added Deadline Event Todo classes, improved command reading technique in main Duke file

Bransonlj committed:
Level-5 Added DukeException class and error handling for bad command inputs

Bransonlj committed:
Level-6 Added delete command functionality

Bransonlj committed:
A-TextUiTesting

Bransonlj committed:
Level-7 adding saving and loading from text file

Bransonlj committed:
Level-8 used LocalDateTime to store date and time for tasks

Bransonlj committed:
Merge branch 'branch-Level-7'

Bransonlj committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

Bransonlj committed:
A-MoreOOP extract classes for more OOP

Bransonlj committed:
A-Packages divide classes into java packages

Bransonlj committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Bransonlj committed:
A-JUnit Add JUnit tests to test the behavior of the code

Bransonlj committed:
A-JavaDoc Add headers to methods

Bransonlj committed:
improve coding style

Bransonlj committed:
Add find command functionality

Bransonlj committed:
Merge branch 'branch-A-JavaDoc'

Bransonlj committed:
Merge branch 'branch-A-CodingStandard'

Bransonlj committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

Bransonlj committed:
Clean up style

Bransonlj committed:
Add GUI

valerietanhx commented on src/main/java/duke/logic/Duke.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    /**
+     * Start the Duke task tracker software.
```
Not sure if this description of the method is accurate; seems more like saving the list of tasks rather than starting up the software?

valerietanhx commented on src/main/java/duke/logic/Storage.java:
```diff
@@ -0,0 +1,106 @@
+            FileReader fileReader = new FileReader(this.filePath);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String line = bufferedReader.readLine();
+            while (line != null) {
```
Hmm maybe the logical units within this while block could be separated better? e.g. lines 57 and 58 could be swapped, and a space could be added between lines 50 and 51.

Descriptions of each logical unit could also be added for clarity!

valerietanhx commented on src/main/java/duke/logic/task/Event.java:
```diff
@@ -0,0 +1,117 @@
+    @Override
+    public String toString() {
+        return "[E]" + super.toString()
+                + " (from: "
```
I like the line breaks here to improve readability! :)

valerietanhx commented on src/main/java/duke/logic/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+        this.isDone = isDone;
+    }
+
+    public boolean getIsDone() {
```
It might be better to name this `isDone()` instead to make it sound more like a boolean method! It also allows linters to automatically verify that this style rule is being followed.

valerietanhx commented on src/main/java/duke/logic/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructor for TaskList object
```
Perhaps this should end in a full stop!

valerietanhx commented on src/main/java/duke/logic/task/Todo.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * Constructor for Todo object with boolean
```
Perhaps this should end in a full stop! :)

valerietanhx commented on src/main/java/duke/logic/task/Todo.java:
```diff
@@ -0,0 +1,69 @@
+     * Constructor for Todo object with boolean
+     *
+     * @param description Description of task.
+     * @param isDone Whether task is marked complete
```
Perhaps this should end in a full stop as well :)

valerietanhx reviewed (commented):
Looks great!! Just a few tiny nitpicks to fix :)

Bransonlj committed:
Add Checkstyle

Bransonlj committed:
Merge branch 'branch-Level-10'

Bransonlj committed:
Merge branch 'A-CheckStyle'

Bransonlj committed:
Add assert

Bransonlj committed:
Refactor code to improve code quality

Use Extract method technique to refactor code in Storage class,
turning fragments of code that can be grouped together into a method
to improve clarity.

Bransonlj committed:
Merge pull request #2 from Bransonlj/branch-A-CodeQuality

Refactor code to improve code quality

Bransonlj committed:
Merge pull request #3 from Bransonlj/branch-A-Assertions

Add assert

Bransonlj committed:
Add schedule command

Add method findDates in TaskList class to find all tasks that occur on given date.
Parser class now parses the schedule command followed by date in format "yyyy-MM-dd",
then prints out a list of all the tasks that occur on the date.

Bransonlj committed:
Merge pull request #4 from Bransonlj/branch-BCD-Extension

Add schedule command

Bransonlj committed:
Give better personality, renamed to Saul

Bransonlj committed:
Add user guide and screenshot

Bransonlj committed:
Clean up some code.

Need to make another commit so the A-FullCommitMessage tag will appear.
Currently its not appearing because i don't have an empty commit with no existing tags.
All of my commits have tags on them so I need to make a new commit so I can put
the A-FullCommitMessage Tag. Thank you!
